src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3VettingHelperTest.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.helpers;
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.et.common.model.ccd.CaseDetails;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.NO;
+
+class Et3VettingHelperTest {
+
+    private CaseDetails caseDetails1;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        caseDetails1 = generateCaseDetails("caseDetailsTest1.json");
+    }
+
+    @Test
+    void givenET3Received_datesShouldShow() {
+        caseDetails1.getCaseData().setClaimServedDate("2022-01-01");
+        String actualResult = Et3VettingHelper.getEt3Dates(caseDetails1.getCaseData());
+        String expectedResult = generateEt3Dates("1 January 2022", "30 January 2022", "1 March 2022");
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void givenET3NotReceived_et3ShouldNotShow() {
+        caseDetails1.getCaseData().setClaimServedDate("2022-01-01");
+        caseDetails1.getCaseData().getRespondentCollection().get(0).getValue().setResponseReceived(NO);
+        String actualResult = Et3VettingHelper.getEt3Dates(caseDetails1.getCaseData());
+        String expectedResult = generateEt3Dates("1 January 2022", "30 January 2022", NO);
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void givenET1isNull_et1ShouldShowError() {
+        caseDetails1.getCaseData().setClaimServedDate(null);
+        String actualResult = Et3VettingHelper.getEt3Dates(caseDetails1.getCaseData());
+        String expectedResult = generateEt3Dates("Cannot find ET1 Served Date", "Cannot find ET3 Due Date", "1 March 2022");
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void givenRespondentCollectionEmpty() {
+        caseDetails1.getCaseData().setRespondentCollection(null);
+        String actualResult = Et3VettingHelper.getEt3Dates(caseDetails1.getCaseData());
+        String expectedResult = generateEt3Dates("Cannot find ET1 Served Date", "Cannot find ET3 Due Date", "No");
+        assertEquals(expectedResult, actualResult);
+    }
+
+    private String generateEt3Dates(String et1ServedDate, String et3DueDate, String et3ReceivedDate) {
+        StringBuilder stringBuilder = new StringBuilder();
```

Comment:
Isn't this code duplication? 
If you did something like this
https://github.com/hmcts/et-ccd-callbacks/blob/f709accbf8bcd50c8a9d9216de29473d3283736b/src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DynamicListHelper.java#L23

Then you could do something like this

https://github.com/hmcts/et-ccd-callbacks/blob/f709accbf8bcd50c8a9d9216de29473d3283736b/src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DynamicListHelperTest.java#L265